### PR TITLE
[CAS-879] Create worker to upload attachments

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-879

### 🎯 Goal

Create worker to upload attachments

### 🧪 Testing

Save attachments to the database:
```
domainImpl.repos.insertAttachmentsToUpload(attachmentsToUpload)
UploadAttachmentsWorker.start(domainImpl.appContext, channelType, channelId, newMessage.id)
```

Observe upload state:
```
domainImpl.scope.launch {
       domainImpl.repos.observeAttachmentsToUpload().distinctUntilChanged().collect {
            ...    
       }
}
```

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/UploadAttachmentsWorker.kt [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        return try {
+            var batchUploaded = true
+
+            attachmentEntities.forEach { entity ->
+                repository.updateAttachmentsToUpload(entity.copy(syncStatus = SyncStatus.IN_PROGRESS))
+
+                val result = uploadAttachment(channelType, channelId, entity)
+                if (result.isSuccess) {
+                    val augmentedEntity = result.data()
+                    repository.updateAttachmentsToUpload(augmentedEntity.copy(syncStatus = SyncStatus.COMPLETED))
```
Comment (samiuelson) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/UploadAttachmentsWorker.kt:58]: Should we remove old entries from the AttachmentsToUpload table somewhere? 🤔

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/AttachmentUploadRepository.kt [hunk 2]
```diff
@@ -22,7 +26,12 @@ internal class AttachmentUploadRepositoryImpl(
         attachmentToUploadDao.updateAttachmentsToUpload(*attachments)
     }
 
-    override suspend fun selectAttachmentsToUploadForMessageId(id: Int): List<AttachmentToUploadEntity> {
-        return attachmentToUploadDao.getAttachmentsToUploadForMessage(id)
+    override suspend fun selectAttachmentsToUploadForMessageId(messageId: String): List<AttachmentToUploadEntity> {
+        return attachmentToUploadDao.getAttachmentsToUploadForMessage(messageId)
+    }
+
+    override fun observeAttachmentsToUpload(): Flow<Map<String, List<AttachmentToUploadEntity>>> {
```
[the remark this answers is not in the document]
Comment (bychkovdmitry) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/AttachmentUploadRepository.kt:33]: I'll rename all the methods

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/UploadAttachmentsWorker.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/UploadAttachmentsWorker.kt:58]: Yes, probably. But how do we know that some upload has failed.

Comment (ogkuzmin) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/UploadAttachmentsWorker.kt:58]: I will remove it in my part of the code `sendMessage`

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/AttachmentUploadRepository.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ogkuzmin) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/attachment/AttachmentUploadRepository.kt:33]: Why do you need this method? It isn't used in your code

**Review by ogkuzmin — COMMENTED** [review]
Discussed not using WorkManager API for this particular task. There is no reason for that.

**Pull request closed without merging**